frontend/src/app/components/containers/complaints/outcomes/hwcr-outcome-by-animal.tsx

```diff
@@ -40,6 +40,19 @@ export const HWCROutcomeByAnimal: FC = () => {
     setShowForm(false);
   };
 
+  const deleteAnimal = (id: number) => {
+    const updatedList = animals.filter(animal => { return animal.id != id
+    });
+    updatedList.forEach(animal => {
+      if(animal.id > id)
+      {
+        animal.id = id;
+      }
+    });
+    setAnimals(updatedList);
+    setShowForm(false);
+  };
+
```

Comment:
I don't think using `filter` is a good idea cuz you want to completely remove it. There are some weird behaviours you may encounter, for ex this bug: add 3 animals -> delete the first one -> delete the second one -> the third one supposed to be there but it's gone as well.

I'm using `splice` instead, it's more straightforward and no need for a second loop through the array.

Also I'm not sure if we need the warning modal popup before deleting, cuz in my ticket it's said there in the requirement, but in yours is not. Should check again with @afwilcox

Reply:
This one I still have to iterate after the filter/splice since I have to reset the key if it's after the key I'm removing is gone - so I don't think there's an advantage over one or the other. My condition to reset the key was animal.id = id; instead of animal.id = animal.id - 1; which was dumb though

Reply:
@cnesmithsalus  - think you should use @Scarlett-Truong's way instead.  It seems to work without needing the iteration.   Also I found another issue with your PR that doesn't exist on hers:

If you press delete while you are adding or editing another one it will close it.   I'm probably going to merge her PR into main first so maybe hold tight...  I've got some more questions to Neil about some of the requirements.